core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -296,107 +296,110 @@ sealed abstract class Parser[+A] {
     * This parser expects the underlying parser to succeed, and will
     * unconditionally backtrack after running it.
     */
-  def peek: Parser[Unit] =
+  def peek: Parser0[Unit] =
     Parser.peek(this)
 
   /** Use this parser to parse between values.
     *
     * Parses `b` followed by `this` and `c`.
     * Returns only the values extracted by `this` parser.
     */
-  def between(b: Parser[Any], c: Parser[Any]): Parser[A] =
+  def between(b: Parser0[Any], c: Parser0[Any]): Parser0[A] =
     (b.void ~ (this ~ c.void)).map { case (_, (a, _)) => a }
 
   /** Use this parser to parse surrounded by values.
     *
     * This is the same as `between(b, b)`
     */
-  def surroundedBy(b: Parser[Any]): Parser[A] =
+  def surroundedBy(b: Parser0[Any]): Parser0[A] =
     between(b, b)
 
   /** Internal (mutable) parsing method.
     *
     * This method should only be called internally by parser instances.
     */
-  protected def parseMut(state: Parser.Impl.State): A
+  private[parse] def parseMut(state: Parser.Impl.State): A
```

Comment:
why does this need to be `private[parse]`, this does weaken the encapsulation a bit.

Reply:
Yes, I still want to try harder to narrow access back to the minimum. This was the easy way out to have a compiling PR showing the external API.

Reply:
If we change this back to `protected` it still works, right?

Reply:
It's now accessed from object Parser, which is a companion of a subclass. Can you do that? IMO spec says yes, compiler says no. https://github.com/scala/bug/issues/2439 says working as intended.